### ISSUE TITLE
web/test_template emits DeprecationWarning

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6326]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)| @tomprince reported|
|-|-|
|Trac ID|trac#6326|
|Type|defect|
|Created|2013-02-24 08:52:09Z|
|Branch|https://github.com/twisted/twisted/tree/web.test_template-emits-DeprecationWarning-6326|

<details><summary>Searchable metadata</summary>

```
trac-id__6326 6326
type__defect defect
reporter__tom_prince tom.prince
priority__normal normal
milestone__ 
branch__branches_web_test_template_emits_DeprecationWarning_6326 branches/web.test_template-emits-DeprecationWarning-6326
branch_author__tomprince tomprince
status__closed closed
resolution__fixed fixed
component__web web
keywords__easy easy
time__1361695929000000 1361695929000000
changetime__1362457149000000 1362457149000000
version__None None
owner__tom_prince tom.prince
cc__jknight
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_2"></a>|@tomprince commented|
|-|-|

(In [37225]) Branching to web.test_template-emits-DeprecationWarning-6326.

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_4"></a>|@wallrj commented|
|-|-|

 * Started [a build](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/web.test_template-emits-DeprecationWarning-6326)
 * See ticket:6323#comment:4

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_5"></a>|@wallrj set owner to @tomprince|
|-|-|

Code Review:

 * Changes look fine.
 * Build looks fine. (in fact better than I've ever seen it!)
 * XMLFile file and path deprecation warnings have disappeared from
   the python 2.6 build results.
   * https://buildbot.twistedmatrix.com/builders/lucid32-py2.6-select/builds/2539/steps/trial/logs/warnings

+1 from me. Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_6"></a>|@tomprince set status to `closed`|
|-|-|

(In [37372]) Merge web.test_template-emits-DeprecationWarning-6326: Filter expected DeprecationWarnings from twisted.web.test.test_template.

Author: tom.prince
Reviewers: rwall
Fixes: [#6326](https://github.com/twisted/twisted/issues/6326)

Passing filenames or files to twisted.web.template.XMLFile is deprecated
but need to be tested. Mask the deprecation warnings until support can be
removed.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra commented|
|-|-|

Minor thingy, make sure the news file is in the correct topfiles folder (should've been web in this case).

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_8"></a>|@wallrj commented|
|-|-|

Ah. Sorry I didn't spot that. Sorry. 
And I was just complaining on IRC that these changes were too small to be worth reviewing individually.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_9"></a>|@tomprince set status to `reopened`|
|-|-|

(In [37381]) Revert r37372: Filter expected DeprecationWarnings from twisted.web.test.test_template.

topfile in the wrong place

Reopens: [#6326](https://github.com/twisted/twisted/issues/6326)

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_10"></a>|@tomprince commented|
|-|-|

(In [37382]) Put the topfile in the right place.

Refs [#6326](https://github.com/twisted/twisted/issues/6326).

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)|@tomprince removed owner<br>@tomprince set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_12"></a>|@thijstriemstra set owner to @tomprince|
|-|-|

Thanks for going through all that trouble tom, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_13"></a>|@tomprince set status to `closed`|
|-|-|

(In [37393]) Merge web.test_template-emits-DeprecationWarning-6326: Filter expected DeprecationWarnings from twisted.web.test.test_template.

Author: tom.prince
Reviewers: rwall, thijs
Fixes: [#6326](https://github.com/twisted/twisted/issues/6326)

Passing filenames or files to twisted.web.template.XMLFile is deprecated
but need to be tested. Mask the deprecation warnings until support can be
removed.
